R2Northstar/NorthstarLauncher
# Adjust for restructured `verified-mods.json`

Updates the launcher code to deal with adjusted verified mods JSON structure defined in https://github.com/R2Northstar/VerifiedMods/pull/41.

The idea here is to allow installing mods from other sources than Thunderstore.

(Somehow) supercedes https://github.com/R2Northstar/NorthstarLauncher/pull/642.

**(should be merged with https://github.com/R2Northstar/VerifiedMods/pull/41!)**

#### TODOs

- [x] Update manifesto format checking
- [x] Wait until manifesto fetching is effectively protected by the MAD flag (https://github.com/R2Northstar/NorthstarLauncher/pull/751)

#### Testing

1. To make your client use the verified mods list's new format, launch Northstar like this: `.\NorthstarLauncher.exe -customverifiedurl=https://raw.githubusercontent.com/R2Northstar/VerifiedMods/refactor/include-mod-source-in-version-links/verified-mods.json`
2. Join mp lobby;
3. Set `allow_mod_auto_download` to 1 through console (`allow_mod_auto_download 1`);
4. `sv_cheats 1`;
5. Fetch manifesto: `script NSFetchVerifiedModsManifesto()`;
6. Download mod: `script NSDownloadMod("nyami.mp_brick", "1.0.2")`;

Looking at your logs, mod should be downloaded normally.
